contracts/contracts/DC.sol

```diff
@@ -0,0 +1,376 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+
+// import {IETHRegistrarController, IPriceOracle} from "@ensdomains/ens-contracts/contracts/ethregistrar/IETHRegistrarController.sol";
+import {IETHRegistrarController, IPriceOracle} from "./IETHRegistrarController.sol";
+
+/**
+    @title A domain manager contract for .country (DC -  Dot Country)
+    @author John Whitton (github.com/johnwhitton), reviewed and revised by Aaron Li (github.com/polymorpher)
+    @notice This contract allows the rental of domains under .country (”DC”)
+    it integrates with the ENSRegistrarController and the ENS system as a whole for persisting of domain registrations.
+    It is responsible for holding the revenue from these registrations for the web2 portion of the registration process,
+    with the web3 regisgtration revenue being held by the RegistrarController contract.
+    An example would be as follows Alice registers alice.com and calls the register function with an amount of 10,000 ONE.
+    5000 ONE would be held by the DC contract and the remaining 5000 funds would be sent to the RegistrarController using 
+    the register function.
+
+ */
+contract DC is Pausable, Ownable {
+    uint256 constant MIN_DURATION = 365 days;
+    bytes constant EMPTY_DATA = bytes("");
+
+    bool public initialized;
+    uint256 public baseRentalPrice;
+    address public revenueAccount;
+    address public registrarController;
+    uint256 public duration;
+    address public resolver;
+    bool public reverseRecord;
+    uint32 public fuses;
+    uint64 public wrapperExpiry;
+
+    IETHRegistrarController registrarControllerContract =
+        IETHRegistrarController(registrarController);
+    IPriceOracle priceOracleContract = IPriceOracle(registrarController);
+
+    // Use a structure for Initial Configuration to fix stack too deep error
+    struct InitConfiguration {
```

Comment:
should use tight layout https://fravoll.github.io/solidity-patterns/tight_variable_packing.html